contracts/src/v0.1/L1EndpointRequestResponse.sol

```diff
@@ -0,0 +1,301 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+import "./RequestResponseConsumerBase.sol";
+import "./L1EndpointBase.sol";
+import "./RequestResponseConsumerFulfill.sol";
+
+abstract contract L1EndpointRequestResponse is
+    RequestResponseConsumerBase,
+    L1EndpointBase,
+    RequestResponseConsumerFulfillUint128,
+    RequestResponseConsumerFulfillInt256,
+    RequestResponseConsumerFulfillBool,
+    RequestResponseConsumerFulfillString,
+    RequestResponseConsumerFulfillBytes32,
+    RequestResponseConsumerFulfillBytes
+{
+    using Orakl for Orakl.Request;
+
+    event DataRequested(uint256 requestId, address sender);
+    event DataRequestFulfilled(
```

Comment:
I think it will be better for the listener to listen to 1 event only. If we split it, we need 6 separate process to listen to 6 different events. It's no good for latency and infrastructure optimization wise.

Reply:
I think 6 is not that much and in case you are supposed to return 0, false, or any other "default" value, you do not know what is the requested data type.

Reply:
We know it through the `JobId` of the `DataRequestFulfilled` event:
```
 event DataRequestFulfilled(
        uint256 requestId,
        uint256 l2RequestId,
        address sender,
        uint256 callbackGasLimit,
        bytes32 jobId,
        uint128 responseUint128,
        int256 responseInt256,
        bool responseBool,
        string responseString,
        bytes32 responseBytes32,
        bytes responseBytes
    );
```

Reply:
Aha! Interesting 🤔